bin/varnishd/cache/cache_fetch.c

```diff
@@ -235,12 +236,20 @@ vbf_stp_retry(struct worker *wrk, struct busyobj *bo)
 	/* VDI_Finish must have been called before */
 	assert(bo->director_state == DIR_S_NULL);
 
+	/* Workspace hasn't been used since rollback */
+	if (bo->ws_bo_rst == WS_Snapshot(bo->ws)) {
```

Comment:
@rezan isn't this silently not executing the rollback if anything else was changed?

Reply:
Right, but you said *rollback* in your previous comment. Those are 2 very different words in the context of this PR.

Reply:
Yes, this is confusing, but the VCL rollback actually is a WS Reset plus restoring the http struct.

So yes, you are right that I should have/could have been more precise.

Thanks for clarifying